pkg/cmd/psa/psa.go

```diff
@@ -0,0 +1,346 @@
+
+	return &cmd
+}
+
+// Validate ensures that all required arguments and flag values are set properly.
+func (o *PSAOptions) Validate() error {
+	if _, ok := validLevels[o.level]; !ok {
+		return fmt.Errorf("invalid level %q", o.level)
+	}
+	return nil
+}
+
+// Complete sets all information required for processing the command.
+func (o *PSAOptions) Complete() error {
+	config := o.configFlags.ToRawKubeConfigLoader()
+	restConfig, err := config.ClientConfig()
+	if err != nil {
+		return fmt.Errorf("failed to create rest config: %w", err)
+	}
+
+	namespace, _, err := config.Namespace()
+	if err != nil {
+		return fmt.Errorf("failed to get namespace: %w", err)
+	}
+	o.namespace = namespace
+
+	// Setup a client with a custom WarningHandler that collects the warnings.
+	o.warnings = &warningsHandler{}
+	restConfig.WarningHandler = o.warnings
+	o.client, err = kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create client: %w", err)
+	}
+
+	if o.printFlags.OutputFormat != nil && len(*o.printFlags.OutputFormat) > 0 {
+		printer, err := o.printFlags.ToPrinter()
+		if err != nil {
+			return err
+		}
+		o.printObj = printer.PrintObj
+
+		return nil
+	}
+
+	o.printObj = func(object runtime.Object, writer io.Writer) error {
+		if err := json.NewEncoder(writer).Encode(object); err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+// Run attempts to update the namespace psa enforce label to the psa audit value.
+func (o *PSAOptions) Run() error {
+	// Get a list of all the namespaces.
+	namespaceList, err := o.getNamespaces()
+	if err != nil {
+		return fmt.Errorf("failed to get namespaces: %w", err)
+	}
+
+	podSecurityViolations := PodSecurityViolationList{}
+	// Gather all the warnings for each namespace, when enforcing audit-level.
+	for _, namespace := range namespaceList.Items {
+		psv, err := o.checkNamespacePodSecurity(&namespace)
+		if err != nil {
+			return fmt.Errorf("failed to check namespace %q: %w", namespace.Name, err)
+		}
+		if psv == nil {
+			continue
+		}
+
+		klog.V(4).Infof(
+			"Pod %q has pod security violations, gathering Pod and Deployment Resources",
+			psv.PodName,
+			namespace.Name,
+		)
+
+		psv.Pod, err = o.client.CoreV1().
+			Pods(namespace.Name).
+			Get(context.Background(), psv.PodName, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf(
+				"failed to get pod %q from %q, which violates %q: %w",
+				psv.PodName,
+				namespace.Name,
+				psv.Violations[0],
+				err,
+			)
+		}
+
+		psv.PodControllers, err = o.getPodControllers(psv.Pod)
+		if err != nil {
+			return fmt.Errorf("failed to get pod controller: %w", err)
+		}
+
+		podSecurityViolations.Items = append(podSecurityViolations.Items, *psv)
+	}
+
+	if len(podSecurityViolations.Items) == 0 {
+		return nil
+	}
+
+	// Print the violations.
+	w := printers.GetNewTabWriter(os.Stdout)
```

Comment:
Why do we need this?, can't we just simply call `o.printObj`(since default printer is already json and we don't require custom printing?)?. If this is necessary, it would be better to pass `IOStreams`(if it exists in this repo) instead `os.Stdout`.